File: c_solutions_1-10/Euler_7.c
#include <stdio.h>
#include <time.h>
#include <math.h>


int is_prime(unsigned long num) {

    unsigned long j;
    unsigned long nroot = sqrt(num) + 1;

    for (j=3; j < nroot; j++) {
        if (num % j == 0) 
            return 0;
    }

    return 1;
}

int main(void) {

    clock_t start, stop;
    start = clock();

    unsigned long curr, cnt;

    for (curr=3, cnt=1; cnt < 10001; curr+=2) {
        if (is_prime(curr))
            cnt++;
    }

    printf ("Answer: %ld\n", curr-2);
    stop = clock();
    printf ("Time: %f\n", ((float)stop - (float)start) / CLOCKS_PER_SEC);

    return 0;
}

File: perl_solutions_1-10/Euler_7.pl
use strict;
use warnings;
use Math::Complex;
use Time::HiRes qw( clock );


sub is_prime
{
    my ($n) = @_;
    if ($n == 2) {
        return 1;
    }
    if ($n % 2 == 0 || $n == 1) {
        return 0;
    }
    for my $i (3..int(sqrt($n)) + 1) {
        if ($n % $i == 0) {
            return 0;
        }
    }
    return 1;
}

sub find_prime
{
    my ($prime_num) = @_;
    my $num_prime = 0;
    my $prime = 1;
    while ($num_prime < $prime_num) {
        $prime++;
        if (is_prime($prime)) {
            $num_prime++;
        }
    }
    return $prime;
}

my $start = clock();
my $answer = find_prime(10001);
my $stop = clock();
print "Answer: ", $answer, "\n";
printf "Time: %.5f\n", $stop - $start;
